nixos/modules/programs/qgroundcontrol.nix

```diff
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.qgroundcontrol;
+in
+{
+
+  options = {
+    programs.qgroundcontrol = {
+      enable = lib.mkEnableOption "qgroundcontrol";
+
+      package = lib.mkPackageOption pkgs "qgroundcontrol" {};
+
+      udevRules = {
+        enable = lib.mkEnableOption "udev rules for qgroundcontrol" // {
+          default = true;
+        };
+
+        rules = lib.mkOption {
+          description = ''
+            udev rules for qgroundcontrol to access serial ports.
+
+            These are quite permissive and may not be suitable for systems with untrusted users.
+            In that case, you may want to adjust these rules to only allow access to specific devices.
+          '';
+          type = lib.types.string;
+          default = ''
+            SUBSYSTEM=="tty", MODE="0666"
```

Comment:
This seems a bit extreme. Wouldn't it be better to somehow let the program be part of the `tty` group? Maybe a sgid wrapper or something?